File: CSCD211 - Programming Principles 2/Lab1/cscd211Utils/FileUtils.java
package cscd211Utils;

import java.util.*;
import java.io.*;

public class FileUtils
{
   public static int countRecords(final Scanner fin, final int linesPer)
   {
      // Precondtions
      if(fin == null || linesPer < 1)
         throw new IllegalArgumentException("bad countRecords param");
         
      int count = 0;
      
      while(fin.hasNext())
      {
         String line = fin.nextLine();
         
         if(!line.isEmpty()) // count goes up only if line is not blank
         {
            fin.nextLine();
            count++;
         }
      }
      
      // Post-Condtions
      if(count == 0 || count%linesPer != 0)
         throw new UnsupportedOperationException("incomplete file");
      
      return count/linesPer;
   }
   
   public static File openInputFile(final Scanner kb)
   {
      // Preconditions
      if(kb == null)
         throw new IllegalArgumentException("bad openInputFile kb param");
         
      String filename = "";
      File inf = null;
      
      do
      {
        System.out.print("Please enter filename: ");
        filename = kb.next();
        kb.nextLine(); // input buffer
        inf = new File(filename);
      } while(!inf.exists() && !inf.canRead());  
      
      return inf;
   }
   
   public static File openInputFile(final String fn)
   {
      // Preconditions
      if(fn == null || fn.isEmpty())
         throw new IllegalArgumentException("bad openInputFile fn param");
         
      File inf = new File(fn);
      
      //Post-conditions
      if(!inf.exists())
         throw new RuntimeException("file doesn't exist");
         
      return inf; 
   }
}

File: CSCD340 - Operating Systems/Lab1/utils/fileUtils.c
#include "fileUtils.h"


int countRecords(FILE * fin, int linesPer)
{
	if(fin == NULL || linesPer < 1)
	{
		perror("Bad params to countRecords\n");
		exit(-99);
	}
	int count = 0;
	char temp[MAX];

	fgets(temp, 100, fin);

	while(!feof(fin))
	{
		count ++;
		fgets(temp, 100, fin);
	}

	if(count == 0 || count % linesPer != 0)
	{
		perror("Bad count in countRecords\n");
		exit(-99);
	}
	
	rewind(fin);

	return count/linesPer;
}


FILE * openInputFile_Prompt()
{
	FILE * fin = NULL;
	char fileName[MAX];
	do
	{
		printf("Please enter the name of the input file ");
		fgets(fileName, MAX, stdin);
		strip(fileName);

		fin = fopen(fileName, "r");

	}while(fin == NULL);

	return fin;
}


void strip(char *array)
{
	if(array == NULL)
	{
		perror("array is null");
		exit(-99);
	}// end if

	int len = strlen(array), x = 0;
   
	while(array[x] != '\0' && x < len)
	{
	  if(array[x] == '\r')
		 array[x] = '\0';

	  else if(array[x] == '\n')
		 array[x] = '\0';

	  x++;

	}// end while
   
}// end strip

